Lab5/src/main/java/agh/ics/oop/WorldMap.java

```diff
@@ -0,0 +1,59 @@
+package agh.ics.oop;
+
+import java.util.List;
+
+public abstract class WorldMap implements IWorldMap {
+
+    protected List<Animal> animalList;
+
+    @Override
+    public List<Animal> getListOfAnimals() {
+        return List.copyOf(this.animalList);
+    }
+
+    @Override
+    public boolean canMoveTo(Vector2d position) {
+        if ( !position.follows(new Vector2d(0,0))) {
+            return false;
+        }
+        // can't use isOccupied, because in grassfield class by isOccupied function we check also, if grass is at this position
```

Comment:
słusznie, ja bym w ogóle tę metodę zostawił tu abstrakcyjną, a zaklepał osobno w każdej z implementacji, bo dość mocno się różnią. Tu też nie ma za bardzo powodu sprawdzać follows(new Vector2d(0,0), bo np. dla trawy to nie ma sensu.